<launch>
    <!-- The chisel server node-->
    <node name="Chisel" pkg="chisel_ros" type="ChiselNode" output="screen"> 
        <!-- Size of the TSDF chunks in number of voxels -->
        <param name="chunk_size_x" value="8"/>
        <param name="chunk_size_y" value="8"/>
        <param name="chunk_size_z" value="8"/>
        <!--- The distance away from the surface (in cm) we are willing to reconstuct -->
        <param name="truncation_scale" value="1.0"/>
        <!-- Whether to use voxel carving. If set to true, space near the sensor will be
             carved away, allowing for moving objects and other inconsistencies to disappear -->
        <param name="use_voxel_carving" value="true"/>
        <!-- When true, the mesh will get colorized by the color image.-->
        <param name="use_color" value="true"/>
        <!-- The distance from the surface (in meters) which will get carved away when
             inconsistencies are detected (see use_voxel_carving)-->
        <param name="carving_dist_m" value="0.05"/>
        <!-- The size of each TSDF voxel in meters-->
        <param name="voxel_resolution_m" value="0.1"/>
        <!-- The maximum distance (in meters) that will be constructed. Use lower values
             for close-up reconstructions and to save on memory. -->
        <param name="far_plane_dist" value="3"/>
        <!-- Mode to use for reconstruction. There are two modes: DepthImage and PointCloud.
             Only use PointCloud if no depth image is available. It is *much* slower-->
        <param name="fusion_mode" value="DepthImage"/>

        <!-- camera type, Pinhole or Fisheye-->
        <param name="camera_type" value= "Fisheye"/>
        <!-- basic information for images-->
        <param name="image_width" value="720"/>
        <param name="image_height" value="720"/>
        <!-- Pinhole Parameter, if not pinhole can be ignored-->
        <!-- <param name="fx" value="367.612580"/> -->
        <!-- <param name="fy" value="369.042951"/> -->
        <!-- <param name="cx" value="369.042951"/> -->
        <!-- <param name="cy" value="236.542114"/> -->
        <!-- Fisheye Parameter, steps a degree, if not can be ignored-->
        <param name="degree_step" value="4"/>

        <!-- Name of the topic -->
        <param name="Depthimage_topic" value="/remode/depth"/>
        <param name="Colorimage_topic" value="/remode/reference"/>
        <param name="Odometry_topic" value="/vins_estimator/cur_pose"/>
    </node>
</launch>